Naming.md

```diff
@@ -46,3 +53,54 @@ protocol FooEventHandling {
     // good
 }
 ```
+
+### Naming a delegate method
+
+The first parameter should always be the item that the delegate was called for.
```

Comment:
Can we say why this is a convention? Add rationale section?

Reply:
Same for other conventions below

Reply:
Good shout. The aim here is to mimic what Apple does for their design and to be consistent.

Reply:
FYI: Apple has this convention for delegate methods in their [Coding Guidelines for Cocoa](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html#//apple_ref/doc/uid/20001282-1001803-BCIDAIJE).